bootstrap/idris2-boot.sh

```diff
@@ -7,22 +7,12 @@ if [ -z "$SCHEME" ]; then
     exit 1
 fi
 
-case $(uname -s) in
-    OpenBSD | FreeBSD | NetBSD)
-        REALPATH="grealpath"
-        ;;
-
-    *)
-        REALPATH="realpath"
-        ;;
-esac
-
-if ! command -v "$REALPATH" >/dev/null; then
-    echo "$REALPATH is required for Chez code generator."
-    exit 1
+if [ "$(uname)" = Darwin ]; then
+    DIR=$(zsh -c 'printf %s "$0:A:h"' "$0")
+else
+    DIR=$(dirname "$(readlink -f -- "$0")")
```

Comment:
Didn't see that, [ref](https://linux.die.net/man/1/readlink). Wonder why that is, what does _preferred_ mean?

Sure, most macOS devs will have package managers, but why make it a dependency just for this? Its use in program output is more of an issue, in case someone wants to deploy a program. Alternatively, you could hardcode the path for fixed deployments.
(a weak argument, but one less `case` has to be a good thing).

Reply:
According to the [GNU Coreutils manual](https://www.gnu.org/software/coreutils/manual/html_node/realpath-invocation.html#realpath-invocation) for realpath(1):
> This is the preferred command for canonicalization as it’s a more suitable and standard name. In addition this command supports relative file name processing functionality.

Neither of those reasons beat `readlink -f`'s ubiquity.